index.d.ts

```diff
@@ -1,13 +1,58 @@
+/**
+ * This file used for adding types to package interfaces that are not documented
+ * by the package (not included in the package's types).
+ */
+
 declare namespace Cypress {
-  import {Log as CtrLog} from "./src/installLogsCollector";
+  import {Log} from "./src/installLogsCollector";
+  import {BuiltinOutputProcessorsTypes} from "./src/types";
 
   interface Cypress {
     TerminalReport: {
-      getLogs<T extends 'txt' | 'json' | 'none' = 'none'>(format?: T): {
-        txt: string,
-        json: string,
-        none: CtrLog[],
-      }[T];
+      getLogs(format: BuiltinOutputProcessorsTypes): string | null;
+      getLogs(format?: 'none' = 'none'): Log[] | null;
     }
+
+    onSpecReady(...args: any[]): void;
+
+    mocha: {
+      getRunner(): Mocha.Runner
+      getRootSuite(): Mocha.Suite
+    }
+
+    state(key: 'done', callback: (error: Error) => Promise<void>): void
+    state(key: 'error', error: Error): void
+  }
+
+  interface Chainable<Subject = any> extends Pick<Promise<Subject>, 'catch'> {
+  }
+}
+
+declare namespace Mocha {
```

Comment:
I think we should do this differently. These types are not part of the public api of cypress-terminal-report. This file should be just for the public global types.

You can however create another d.ts that we can include specifically in tsconfig for types, to be used only internally.

Reply:
That's a great point, I didn't realize this file is included in the package. Moved these new ones to a separate file that I added to `.npmignore`: https://github.com/archfz/cypress-terminal-report/pull/261/commits/bf84798fc0245fa0e58d57fecd992cc2429ab1ee